styx-api-service/src/main/java/com/spotify/styx/api/workflow/WorkflowInitializer.java

```diff
@@ -45,7 +45,8 @@ public WorkflowInitializer(Storage storage, Time time) {
     this.time = Objects.requireNonNull(time);
   }
 
-  public Optional<Workflow> inspectChange(Workflow workflow) {
+  public Optional<Workflow> inspectChangeAndStore(Workflow workflow)
```

Comment:
Nitpick but the naming here is confusing me. Isn't the primary purpose of this method to handle storing a (new) workflow? Calculating the new next natural trigger is done as well, as a side effect part of storing the workflow. I wouldn't call that "inspecting" and I can't see any other inspection going on. How about just calling this method `storeWorkflow` ?

Reply:
That makes sense. I don't know why it was named as `inspect`. :)